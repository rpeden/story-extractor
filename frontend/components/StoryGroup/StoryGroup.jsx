import React from "react";
import Paper from "material-ui/Paper";
import _ from "lodash";

import Story from "../Story/Story.jsx";

const StoryGroup = ({storyGroup}) => {
    const sortedStories = _.orderBy(storyGroup.stories, ["points"], ["desc"]);
    const topStories = _.take(sortedStories, 10);
    
    return (
        <Paper key={storyGroup.site} zDepth={2} className="story-group-container">
          <div className="story-group-inner">
            <div className="story-group-header">
              <img className="story-group-logo" src={`/logos/${storyGroup.logo}`} />
              <span className="story-group-site-name">{storyGroup.site}</span>
            </div>
            <div className="story-group-points">Points</div>
            <div className="story-group-title">Title</div>
            <div className="story-group-stories-container">
              {topStories.map((story) => <Story story={story} /> )}
            </div>
          </div>
        </Paper>
    );
}

export default StoryGroup;