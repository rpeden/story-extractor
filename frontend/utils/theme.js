import getMuiTheme from "material-ui/styles/getMuiTheme";
import { red500, red700 } from "material-ui/styles/colors";
import injectTapEventPlugin from "react-tap-event-plugin";

injectTapEventPlugin();

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: red500,
    primary2Color: red700
  }
});

export default muiTheme;
