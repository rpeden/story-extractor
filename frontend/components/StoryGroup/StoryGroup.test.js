import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import theme from "../../utils/theme";

import StoryGroup from "./StoryGroup.jsx";

const testStories = [
  {
    title: "Test Title",
    points: 50,
    address: "http://nowhere.com"
  },
  {
    title: "Test Two",
    points: 55,
    address: "http://nowhere2.com"
  },
  {
    title: "Test Three",
    points: 60,
    address: "http://nowhere3.com"
  }
];

const storyGroupData = {
  stories: testStories,
  site: "Test Site",
  logo: "nothing.png"
};

test("Story group snapshot", () => {
  const component = renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <StoryGroup storyGroup={storyGroupData} />
    </MuiThemeProvider>
  );
  const storyGroup = component.toJSON();
  expect(storyGroup).toMatchSnapshot();
});

test("Site name displays correctly", () => {
  const component = shallow(
      <StoryGroup storyGroup={storyGroupData} />
  );

  const storyName = component.find(".story-group-site-name").text();
  expect(storyName).toBe(storyGroupData.site);
});

test("Correct group logo is used", () => {
  const component = shallow(
      <StoryGroup storyGroup={storyGroupData} />
  );

  const groupLogoSrc = component.find(".story-group-logo").prop("src");
  expect(groupLogoSrc).toMatch(storyGroupData.logo);
});
