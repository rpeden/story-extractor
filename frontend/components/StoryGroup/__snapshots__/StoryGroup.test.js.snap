// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story group snapshot 1`] = `
<div
  className="story-group-container"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
             0 3px 10px rgba(0, 0, 0, 0.23)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div
    className="story-group-inner"
  >
    <div
      className="story-group-header"
    >
      <img
        className="story-group-logo"
        src="/logos/nothing.png"
      />
      <span
        className="story-group-site-name"
      >
        Test Site
      </span>
    </div>
    <div
      className="story-group-stories-container"
    >
      <div
        style={
          Object {
            "borderTop": "1px solid #ccc",
            "marginBottom": "10px",
            "paddingTop": "10px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "block",
            }
          }
        >
          <a
            href="http://nowhere3.com"
            onClick={[Function]}
          >
            Test Three
          </a>
        </div>
        <div
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
              "width": "150px",
            }
          }
        >
          60
           points
        </div>
        <span
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
            }
          }
        >
           (
          nowhere3.com
          )
        </span>
      </div>
      <div
        style={
          Object {
            "borderTop": "1px solid #ccc",
            "marginBottom": "10px",
            "paddingTop": "10px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "block",
            }
          }
        >
          <a
            href="http://nowhere2.com"
            onClick={[Function]}
          >
            Test Two
          </a>
        </div>
        <div
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
              "width": "150px",
            }
          }
        >
          55
           points
        </div>
        <span
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
            }
          }
        >
           (
          nowhere2.com
          )
        </span>
      </div>
      <div
        style={
          Object {
            "borderTop": "1px solid #ccc",
            "marginBottom": "10px",
            "paddingTop": "10px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "block",
            }
          }
        >
          <a
            href="http://nowhere.com"
            onClick={[Function]}
          >
            Test Title
          </a>
        </div>
        <div
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
              "width": "150px",
            }
          }
        >
          50
           points
        </div>
        <span
          style={
            Object {
              "color": "#888",
              "display": "inline-block",
              "fontSize": "13px",
            }
          }
        >
           (
          nowhere.com
          )
        </span>
      </div>
    </div>
  </div>
</div>
`;
